File: Easy/Java/Best_Time_to_Buy_and_Sell_Stock_II(122).java
class Solution {
    public int maxProfit(int[] prices) {
        int held = Integer.MIN_VALUE, sold = 0;
        for (int price : prices) {
            int preSold = sold;
            sold = Math.max(sold, held + price);
            held = Math.max(held, preSold - price);
        }
        return sold;
    }
}

File: Easy/Python 3/Best_Time_to_Buy_and_Sell_Stock_II(122).py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        total = 0
        buy = prices[0]
        
        for p in prices:
            if p < buy:
                buy = p
            elif p > buy:
                total += p - buy
                buy = p
                
        return total